prow/resultstore/uploader.go

```diff
@@ -35,13 +35,19 @@ func NewUploader(client *Client) *Uploader {
 
 // Upload uploads a completed Prow job's results to ResultStore's API:
 // https://github.com/googleapis/googleapis/blob/master/google/devtools/resultstore/v2/resultstore_upload.proto
+// If error is returned, the call should be retried later.
 func (u *Uploader) Upload(ctx context.Context, log *logrus.Entry, p *Payload) error {
 	inv, err := p.Invocation()
 	if err != nil {
-		return err
+		log.Errorf("p.Invocation: %v", err)
+		return nil
 	}
 	w, err := writer.New(ctx, log, u.client, inv)
 	if err != nil {
+		if writer.PermanentError(err) {
```

Comment:
FTR the fact that we were checking the error from `writer.New()` for PermanentError-ness was not obvious to me. I assumed that the error was for the actual uploading process (which seem to be the various WriteConfiguration(), WriteTarget(), etc calls on lines 53-57 below). Are we purposely doing "fire and forget" for those calls because we don't want to retry them even if any of them fail? Perhaps it's worth adding a comment there explaining our current posture WRT those unchecked errors?